uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -1112,12 +1112,236 @@ public function testFindOverlappingLeaveRequestsForOneOverlappingLeaveRequest()
+        'to_date' => $testToDate,
+        'type_id' => $absenceTypeForTOILAccrual->id,
+        'request_type' => $testingRequestType
+      ]);
+
+      // Flush leave request from DB to get ready for the next test suite
+      $leaveRequestToTest->delete();
+
+      $this->assertCount(0, $overlappingRequests);
+    }
+  }
+
+  public function testFindOverlappingLeaveRequestsForAMPMRequests() {
+    $contactID = 1;
+
+    $testFromDate = '2018-04-13 00:00:00';
+    $testToDate = '2018-04-15 23:59:00';
+    $halfDayAMID = $this->leaveRequestDayTypes['half_day_am']['value'];
+    $halfDayPMID = $this->leaveRequestDayTypes['half_day_pm']['value'];
+
+    $absenceType = AbsenceTypeFabricator::fabricate([ 'calculation_unit' => 1 ]);
+
+    // Test suites: date from, date from type, date to, date to type, should overlap or not
+    $overlappingRequestsTestSuites = [
+      ['2018-04-11', 1, '2018-04-13', $halfDayAMID, FALSE],
+      ['2018-04-11', 1, '2018-04-13', $halfDayPMID, TRUE],
+      ['2018-04-11', 1, '2018-04-13', 1, TRUE],
+      ['2018-04-15', 1, '2018-04-17', 1, TRUE],
+      ['2018-04-15', $halfDayAMID, '2018-04-17', 1, TRUE],
+      ['2018-04-15', $halfDayPMID, '2018-04-17', 1, FALSE]
+    ];
+
+    foreach ($overlappingRequestsTestSuites as $overlappingRequestsTestSuite) {
+      $requestFromDate = new DateTime($overlappingRequestsTestSuite[0]);
+      $requestFromDateType = $overlappingRequestsTestSuite[1];
+      $requestToDate = new DateTime($overlappingRequestsTestSuite[2]);
+      $requestToDateType = $overlappingRequestsTestSuite[3];
+      $shouldRequestOverlap = $overlappingRequestsTestSuite[4];
+
+      $leaveRequestToTest = LeaveRequestFabricator::fabricateWithoutValidation([
+        'type_id' => $absenceType->id,
+        'contact_id' => $contactID,
+        'from_date' => $requestFromDate->format('YmdHis'),
+        'from_date_type' => $requestFromDateType,
+        'to_date' => $requestToDate->format('YmdHis'),
+        'to_date_type' => $requestToDateType
+      ], true);
+
+      $overlappingRequests = LeaveRequest::findOverlappingLeaveRequests([
+        'contact_id' => $contactID,
+        'from_date' => $testFromDate,
+        'from_date_type' => $halfDayPMID,
+        'to_date' => $testToDate,
+        'to_date_type' => $halfDayAMID,
+        'type_id' => $absenceType->id,
+        'request_type' => LeaveRequest::REQUEST_TYPE_LEAVE
+      ]);
+      $leaveRequestToTestID = $leaveRequestToTest->id;
+
+      // Flush leave request from DB to get ready for the next test suite
+      $leaveRequestToTest->delete();
+
+      $this->assertCount($shouldRequestOverlap ? 1 : 0, $overlappingRequests);
+      if ($shouldRequestOverlap) {
+        $this->assertInstanceOf(LeaveRequest::class, $overlappingRequests[0]);
+        $this->assertEquals($overlappingRequests[0]->id, $leaveRequestToTestID);
+      }
+    }
+  }
+
+  public function testFindOverlappingLeaveRequestsTOILinDaysTreatedAsRequestInHours() {
```

Comment:
This is a generic test name, at first glance for someone not familiar with the code, the name provokes another question, How are request in hours supposed to be treated?  
This generic name allows a lot of things that could have been splitted into different tests to be bundled as one.